doc/how_to/dataclass/pydantic.md

```diff
@@ -0,0 +1,258 @@
+# Interact with Pydantic
+
+This how-to guide demonstrates how to easily enable interaction with [Pydantic](https://docs.pydantic.dev/latest/) using familiar APIs from Panel and [Param](https://param.holoviz.org/), such as `watch`, `bind`, `depends`, and `rx`.
+
+## Overview
+
+The `pn.dataclass` module provides functions to synchronize the fields of Pydantic `BaseModel` with a Parameterized object, a Panel widget, or an `rx` value. It also provides a `ModelViewer` class for creating a Layoutable Viewer that wraps an Pydantic `BaseModel` class or instance.
+
+### Terminology
+
+- **Pydantic**: A library for creating dataclass like models (`BaseModel`) with non-observable fields.
+- **model**: Refers to Pydantic `BaseModel` classes and instances.
+- **names**: Refers to the names of the fields/parameters to synchronize. Can be an iterable or a dictionary mapping from field names to parameter names.
+
+### Classes
+
+- **`ModelParameterized`**: An abstract Parameterized base class that can wrap a Pydantic `BaseModel` class or instance.
+- **`ModelViewer`**: An abstract base class for creating a Layoutable Viewer that can wrap a Pydantic `BaseModel` class or instance.
+
+### Functions
+
+- **`create_rx`**: Can create `rx` values from fields of a model, synced to the fields of the model.
+- **`sync_with_parameterized`**: Syncs the fields of a model with the parameters of a Parameterized object.
+- **`sync_with_widget`**: Syncs a field of the model with the value of a Panel widget.
+- **`sync_with_rx`**: Syncs a field of a model with an `rx` value.
+
+All synchronization is from Parameterized to Pydantic model. The other way is currently not supported.
+
+Only top-level fields/parameters are synchronized, not nested ones.
+
+## Synchronize a Field of a model with a Panel Widget
+
+Use `sync_with_widget` to synchronize a field of a model with the `value` parameter of a Panel widget.
+
+```{pyodide}
+import panel as pn
+from datetime import datetime
+from pydantic import BaseModel
+import json
+
+pn.extension()
+
+
+class DeliveryModel(BaseModel):
+    timestamp: datetime = datetime(2021, 1, 1, 1, 1, 1)
+
+model = DeliveryModel()
+
+widget = pn.widgets.DatetimeInput(name="Timestamp")
+pn.dataclass.sync_with_widget(model, widget, "timestamp")
+
+def view_model(*args):
+    return json.loads(model.json())
+
+pn.Column(
+    widget,
+    pn.pane.JSON(pn.bind(view_model, widget)),
+).servable()
+```
+
+Todo: Fix order of updates. The JSON pane updates before the model is updated.
```

Comment:
For some unknown reason the json pane updates before the model. A workaround is to make `view_model` async and wait 250ms.

This should be understood and fixed.

Reply:
Fixed